src/renderer/shader.cpp

```diff
@@ -260,7 +260,7 @@ static void uniform(lua_State* L, const char* name, const char* type)
 static void pass(lua_State* state, const char* name)
 {
 	auto* cmb = getCombinations(state);
-	copyString(cmb->m_passes[cmb->m_pass_count], name);
+	copyString(cmb->m_passes[cmb->m_pass_count].data, name);
```

Comment:
It is a consequence of changing Pass from `char[20]` to `StaticString<20>`. `copyString` takes static-sized C array or pointer and length.
Original errors were located in `array.h` when using `Array<Pass>` like that:

```
In file included from ../../../src/renderer/renderer.cpp:3:0:
../../../src/engine/core/array.h: In instantiation of ‘T& Lumix::Array<T>::emplace(Params&& ...) [with Params = {const char*&}; T = char [32]]’:
../../../src/renderer/renderer.cpp:746:34:   required from here
../../../src/engine/core/array.h:222:3: error: no matching function for call to ‘operator new [](sizetype, Lumix::NewPlaceholder, char*)’
   new (NewPlaceholder(), (char*)(m_data + m_size)) T(myforward<Params>(params)...);
   ^
../../../src/engine/core/array.h:222:3: note: candidate is:
<built-in>:0:0: note: void* operator new [](long unsigned int)
<built-in>:0:0: note:   candidate expects 1 argument, 3 provided
../../../src/engine/core/array.h: In instantiation of ‘void Lumix::Array<T>::callDestructors(T*, T*) [with T = char [32]]’:
../../../src/engine/core/array.h:90:42:   required from ‘Lumix::Array<T>::~Array() [with T = char [32]]’
../../../src/renderer/renderer.cpp:623:26:   required from here
../../../src/engine/core/array.h:364:4: error: request for member ‘~char [32]’ in ‘* begin’, which is of non-class type ‘char [32]’
    begin->~T();
    ^

```